packages/hash/frontend/src/components/GlideGlid/glide-grid.tsx

```diff
@@ -103,6 +119,31 @@ const GlideGrid: ForwardRefRenderFunction<DataEditorRef, DataEditorProps> = (
     [onVisibleRegionChanged],
   );
 
+  const handleColumnResize = useCallback(
+    (column: GridColumn, newSize: number) => {
+      setColumns((prevCols) => {
+        const index = prevCols.findIndex(({ title }) => title === column.title);
+        const newColumns = [...prevCols];
+
+        newColumns.splice(index, 1, {
+          ...prevCols[index]!,
+          width: newSize,
+        });
+        return newColumns;
+      });
+    },
+    [],
+  );
+
+  if (
+    !isEqual(
+      columnsProperty.map((col) => col.title),
+      columns.map((col) => col.title),
+    )
+  ) {
+    setColumns(columnsProperty);
+  }
```

Comment:
If you're going to keep this approach of storing the entire columns in state (read above) then: 

You should keep `columnsProperty` in a `prevColumnsProperty` state instead to do this more performantly / make this easier to read – and just require the consumer keep `columns` prop stable between renders, which from the usages I've checked, they do. 

```suggestion
  const [prevColumnsProperty, setPrevColumnsProperty] = useState(columnsProperty);
  if (prevColumnsProperty !== columnsProperty) {
    setColumns(columnsProperty);
    setPrevColumnsProperty(columnsProperty);
  }
```